internal/providers/provider.go

```diff
@@ -522,6 +537,9 @@ type ImportResourceStateRequest struct {
 
 	// ClientCapabilities contains information about the client's capabilities.
 	ClientCapabilities ClientCapabilities
+
+	// Identity is the identity data of the resource.
+	Identity cty.Value
 }
 
 type ImportResourceStateResponse struct {
```

Comment:
This is the plan we were talking about changing the protocol, either with a new call, or a new return field.

Also something to keep in mind, extending the protocol with a corresponding capability flag would also allow the provider to indicate if something does not exist (as opposed to a generic error), helping with the conditional import problem. That's not an idea we we need to block on for this portion of the resource identity changes, but we don't want to rule it out.